words2num/lang_EN_US.py

```diff
@@ -140,6 +140,8 @@ def compute_placevalues(tokens):
 
 def tokenize(text):
     tokens = re.split(r"[\s,\-]+(?:and)?", text.lower())
+    # Remove empty strings caused by split
+    tokens = [tok for tok in tokens if tok]
```

Comment:
Prefer lazy evaluation, in general.
```suggestion
    tokens = (tok for tok in tokens if tok)
```

Reply:
That won't work in this case because we'll consume the generator in `compute_placevalues()` before we get to the rest of the `tokenize()` logic.